app/src/test/java/ch/hikemate/app/model/facilities/FacilitiesRepositoryTest.kt

```diff
@@ -0,0 +1,334 @@
+              assertNotEquals(routes.size, 0)
+              print(routes)
+            }) {
+              fail("Should not fail")
+            }
+      }
+
+  @Test
+  fun testGetFacilities_callsClient() {
+    `when`(client.newCall(any())).thenReturn(mock())
+    facilitiesRepository.getFacilities(
+        testBoundsNormal,
+        { routes ->
+          assertNotEquals(routes.size, 0)
+          print(routes)
+        }) {
+          fail("Failed to fetch from Overpass API")
+        }
+    verify(client).newCall(any())
+  }
+
+  @Test
+  fun testGetFacilities_andFilterAmenitiesWithNormalInput() {
+
+    val response =
+        buildResponse(VALID_RESPONSE_HEADER + VALID_RESPONSE_ELEMENT + VALID_RESPONSE_FOOTER)
+
+    val callbackCaptor = argumentCaptor<okhttp3.Callback>()
+    val latch = CountDownLatch(1)
+
+    setupMockResponse(response)
+
+    facilitiesRepository.getFacilities(
+        bounds = testBoundsNormal,
+        onSuccess = { facilities ->
+          assertEquals(1, facilities.size)
+          assertEquals(FacilityType.TOILETS, facilities[0].type)
+          latch.countDown()
+          Log.d(
+              "FacilitiesRepositoryTest",
+              "getFacilities with normal parameters succeeded. Got: ${facilities.size} facilities")
+        },
+        onFailure = { fail("getFacilities call failed") })
+
+    verify(mockCall).enqueue(callbackCaptor.capture())
+
+    assert(latch.await(5, TimeUnit.SECONDS)) { "Test timed out" }
+  }
+
+  @Test
+  fun testGetFacilitiesWithTruncatedResponse() {
+    val response =
+        buildResponse(
+            VALID_RESPONSE_HEADER +
+                VALID_RESPONSE_ELEMENT +
+                """
+            {
+            "type": "node",
+            "id": 992442027,
+            "lat": 46.5098337,
+            """ // This node is truncated
+                +
+                VALID_RESPONSE_FOOTER)
+
+    setupMockResponse(response)
+
+    val failureCounter = AtomicInteger(0)
+
+    facilitiesRepository.getFacilities(
+        bounds = testBoundsNormal,
+        onSuccess = { _ -> fail("Should fail") },
+        onFailure = { _ ->
+          Log.d(
+              "FacilitiesRepositoryTest",
+              "getFacilities with truncated response failed as expected")
+          failureCounter.incrementAndGet()
+        })
+
+    verify(mockCall).enqueue(callbackCaptor.capture())
+    assertEquals(1, failureCounter.get())
+  }
+
+  @Test
+  fun testGetFacilitiesWithEmptyResponse() {
+    val response =
+        buildResponse(
+            VALID_RESPONSE_HEADER +
+                VALID_RESPONSE_FOOTER) // There are no nodes returned in this response
+
+    setupMockResponse(response)
+
+    facilitiesRepository.getFacilities(
+        bounds = testBoundsNormal,
+        onSuccess = { facilities -> assertEquals(0, facilities.size) },
+        onFailure = { _ -> fail("getFacilities call with empty response failed") })
+
+    verify(mockCall).enqueue(callbackCaptor.capture())
```

Comment:
Warning: same remark as above. Please check the the callback is called with a boolean

Reply:
done